tutorials/use-cloud-pubsub-cloud-storage-app-engine/index.md

```diff
@@ -0,0 +1,1314 @@
+
+1. Choose a directory to house your project. From this point forward, this will be referred to as the host directory. Inside your host directory, create a new directory called `lib` for the storage of external libraries.
+1. In your host directory, create the file `requirements.txt` and copy in the following text:
+
+    ```txt
+    jinja2==2.9.6
+    webapp2==3.0.0b1
+    GoogleAppEngineCloudStorageClient==1.9.22.1
+    google-api-python-client==1.6.2
+    ```
+
+    This specifies which libraries are necessary for your application.
+
+1. Install the external libraries into your `lib` directory:
+
+    ```sh
+    pip install -t lib -r requirements.txt
+    ```
+
+1. In your host directory, create the file `appengine_config.py` and copy in the following code:
+
+    ```py
+    from google.appengine.ext import vendor
+    vendor.add('lib')
+    ```
+
+1. In your host directory, create an `app.yaml` file and copy in the following code:
+
+    ```yaml
+    runtime: python27
+    api_version: 1
+    threadsafe: yes
+
+    handlers:
+    - url: /_ah/push-handlers/.*
+      script: main.app
+      login: admin
+
+    - url: .*
+      script: main.app
+    ```
+
+### HTML files
+
+The HTML files represent the different pages of your web application.
+
+1. In your host directory, create a `templates` directory to hold all of your HTML files. Each HTML file should have the same basic layout, with a title and links to the other pages of your application:
+
+    ```html
+    <!DOCTYPE html>
+    <html>
+
+      <head>
+        <title>Page Title</title>
+      </head>
+
+      <body>
+
+        <ul class="links">
+          <li><a href="/">Home</a></li>
+          <li><a href="/photos">Photos</a></li>
+          <li><a href="/search">Search</a></li>
+        </ul>
+
+        <h1>Page Title</h1>
+
+      </body>
+    </html>
+    ```
+
+1. Create an HTML file for the home/notifications page of your application (url: `http://[PROJECT ID].appspot.com`) using the template given above. The notifications page will have a news feed listing all recent actions performed on your GCS photo bucket.
+1. Create an HTML file for the photos page of your application (url: `http://[PROJECT ID].appspot.com/photos`) using the template given above. The photos page will display the thumbnails and names of all photos uploaded to your GCS photo bucket.
+1. Create an HTML file for the search page of your application (url: `http://[PROJECT ID].appspot.com/search`) using the template given above. The search page will display the thumbnails and names of the photos uploaded to your GCS photo bucket that match the entered search term.
+
+### The `main.py` file
+
+The `main.py` file contains the backend logic of the website, including the reception of Cloud Pub/Sub messages, the communication with GCS and Datastore, and the rendering of HTML templates.
```

Comment:
I'd change this to "The `main.py` file contains the backend logic of the website, including sending and receiving Cloud Pub/Sub messages,  communicating with GCS and Datastore, and rendering HTML templates."  to make it more active.